platform/admin/core/admin-core-alertservice/src/main/java/org/codice/ddf/admin/core/alert/service/internal/AlertServiceBean.java

```diff
@@ -0,0 +1,202 @@
+        if (objectName != null && mBeanServer != null) {
+            unregisterMBean();
+            LOGGER.debug("Unregistered [{}] MBean.", CLASS_NAME);
+        }
+    }
+
+    @Override
+    @Nonnull
+    public List<Map<String, Object>> getAlerts() {
+        List<Map<String, Object>> currentAlerts = new ArrayList<>();
+
+        // add InsecureDefaultsService alerts
+        List<String> insecureDefaultsAlertDetails = insecureDefaultsService.validate()
+                .stream()
+                .map(org.codice.ddf.admin.insecure.defaults.service.Alert::getMessage)
+                .collect(Collectors.toList());
+        if (!insecureDefaultsAlertDetails.isEmpty()) {
+            final Alert insecureDefaultsAlert = new Alert(Alert.Level.ERROR,
+                    INSECURE_DEFAULTS_ALERT_TITLE,
+                    insecureDefaultsAlertDetails);
+            currentAlerts.add(insecureDefaultsAlert);
+        }
+
+        // add alerts from the EventService
+        currentAlerts.addAll(addedAlerts.entrySet()
+                .stream()
+                .map(alertEntry -> {
+                    final Alert alert = alertEntry.getValue();
+                    alert.put("key", alertEntry.getKey());
+                    return alert;
+                })
+                .collect(Collectors.toList()));
+
+        return currentAlerts;
+    }
+
+    @Override
+    @Nonnull
+    public Boolean dismissAlert(@Nonnull String key) {
+        final Alert dismissedAlert = addedAlerts.remove(key);
+        if (dismissedAlert == null) {
+            LOGGER.debug("Failed to dismiss alert with key {}", key);
+            return false;
+        } else {
+            LOGGER.debug("Dismissed alert with title=\"{}\" and key=\"{}\"",
+                    dismissedAlert.getTitle(),
+                    key);
+            return true;
+        }
+    }
+
+    /**
+     * Adds an {@link Alert} to the collection of current {@link Alert}s. The even must have topic
+     * {@link Alert#ALERT_EVENT_TOPIC_VALUE} and
+     * event.getProperty({@link Alert#ALERT_EVENT_ALERT_PROPERTY}) is the {@link Alert} to add.
+     * Already-added {@link Alert}s with the same {@value Alert#KEY} as the {@link Alert} in the event will be
+     * overriden.
+     *
+     * @param event
+     */
+    @Override
+    public void handleEvent(Event event) {
+        if (event != null && ALERT_EVENT_TOPIC_VALUE.equals(event.getTopic())) {
+            final Object alertObject = event.getProperty(ALERT_EVENT_ALERT_PROPERTY);
+            if (alertObject instanceof Alert) {
+                final Alert alert = (Alert) alertObject;
+                String key = alert.getKey();
+
+                // added {@link Alert}s must have a {@value Alert#KEY} because they must be dismissable
+                if (key == null) {
+                    key = UUID.randomUUID()
+                            .toString()
+                            .replaceAll("-", "");
+                    alert.setKey(key);
```

Comment:
See response above to your question "What's the thinking behind overriding alerts with the same key? Does it serve a purpose of fulfill a use case? Just asking."

Basically, using this key allows me to not have to implement an equals method for Alert to check if the alert already exists in the map before adding.